### PR TITLE
Features/james/chat

### PR DESCRIPTION
**Pull request by villanuevajamesfvillanueva**
1. Added temporary sidebar in home index that displays current user's conversations
image
2. Each conversation links to its show page; 
    p.s.  temporary sidebar should also be in the show page
image
3. Added a send message button current user is looking at another user's item
image
4. Clicking 'Send Message' button creates a new conversation on the item between the user that posted the item and the inquiring user, if the conversation doesn't exist yet
5. temporary sidebar and 'Send Message' button don't appear for users that are not logged in

### CODE REVIEW

**Review by Eunicegenel — APPROVED** [review]

**Review by Eunicegenel — COMMENTED** [review]
pa add new line boss dito -> app/assets/stylesheets/home.scss

**Review by Eunicegenel — CHANGES_REQUESTED** [review]
eto din boss pa add new line -> app/views/conversations/show.html.erb

**Review by Eunicegenel — CHANGES_REQUESTED** [review]
eto din boss, newline -> app/views/home/_sidebar.html.erb

File: app/models/conversation.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@ class Conversation < ApplicationRecord
   has_many :messages, class_name: 'Message', dependent: :destroy
   belongs_to :user1, class_name: 'User'
   belongs_to :user2, class_name: 'User'
+  belongs_to :item, class_name: 'Item'
```
Comment (analyn-cajocson) [inline app/models/conversation.rb]: Do we really need to define the `class_name`?

File: app/models/item.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@ class Item < ApplicationRecord
   validates :name, presence: true
   validates :status, presence: true
 
-  belongs_to :user
+  belongs_to :user, class_name: 'User'
```
Comment (analyn-cajocson) [inline app/models/item.rb]: Why do we have to define the `class_name` here? Are you planning to add another model association for the `User` model?

File: app/views/home/index.html.erb [hunk 3]
```diff
@@ -4,12 +4,17 @@
...
+
       <div class="post_here">
         <button type="button" class="btn btn-primary down" data-toggle="modal" data-target="#exampleModal">
           Post New Item
         </button>
       </div>
     <% end %>
+
+    
+
```
Comment (analyn-cajocson) [inline app/views/home/index.html.erb]: Remove excess new line

File: app/views/items/show.html.erb [hunk 4]
```diff
@@ -2,12 +2,19 @@
   <%= stylesheet_link_tag "items" %> 
 </head> 
 <body>
+    <% if user_signed_in? %>
+        <%= render 'home/sidebar' %>
+    <% end %>
+
```
Comment (analyn-cajocson) [inline app/views/items/show.html.erb:8]: Remove excess new line

**Review by leocsdev — APPROVED** [review]

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

**Review by analyn-cajocson — APPROVED** [review]

**Review by Eunicegenel — APPROVED** [review]

**Pull request merged**